src/engine/source/indexerconnector/include/indexerConnector/indexerConnector.hpp

```diff
@@ -98,6 +101,78 @@ class EXPORTED IndexerConnector final : public IIndexerConnector
      * @copydoc IIndexerConnector::publish
      */
     void publish(const std::string& message) override;
+
+    /**
+     * @brief Auxiliary class to create the queries that will be sent to the indexer.
+     *
+     */
+    class QueryBuilder final : public base::utils::patterns::Builder<QueryBuilder>
```

Comment:
1- Let's remove this from the include, this is not part of what we want to expose with the library. Create a new class.

2- I think there is too much abstraction for this, it seems better to implement two classes.


1. To delete
2. For bulkindex

The delete one has as input the index and the id, therefore you could build the string in the build method using string_views that are attributes of the class.

For bulkindex, it has 3 input data, the index, the id and data. Since id is optional, you could take the if to the build() method.

It would look like this:
bulkData.append(BulkIndexBuilder::builder()
                            .index(index)
                            .id(id)
                            .rawData(data)
                            .build());
                            
bulkData.append(DeleteQueryBuilder::builder()
                        .index("_index", index)
                        .id("_id", id)
                        .build());

Reply:
Hi @Dwordcito 

1- I created a nested class because the query builder didn't seem generic enough to be an util. I can declare the classes at `indexerConnector.cpp`

2- If I create a class that is specific to build an insert/delete query, the builder pattern might not be the most convenient choice then in my opinion. A simple class with a constructor and default parameters would do the trick. 

This is because I need to allow the user to change the order of the methods or omit some, from the outside all of them should be valid

```c++

bulkData.append(BulkIndexBuilder::builder().index(index).id(id).rawData(data).build());
bulkData.append(BulkIndexBuilder::builder().id(id).index(index).rawData(data).build());
bulkData.append(BulkIndexBuilder::builder().rawData(data).index(index).id(id).build());
bulkData.append(BulkIndexBuilder::builder().rawData(data).index(index).build());
```

To cover this situation, I can't construct the string appending the result of each method but saving each value until the end (rawData for example is out of the JSON and adds new line characters).

The resulting "builder" would be really similar to a simple `bulkData.append(BulkIndexBuilder(index,id,rawData)` that already has the two types of possible resulting queries stored and replaces the placeholders.  

Another option is to simply rollback all the approach and make a tiny change at `builderBulkIndex` using the same handcrafted mechanism that is implemented right now. This would be straightforward.

Reply:
1- No, you must give it enough entity and create a new header if necessary.

2- Talk to your TL if you want to change the approach, my suggestions are, but as it stands I won't approve it.

Reply:
After discussing this in the Daily we concluded to aim for the minimalistic approach

Even though we think a proper query-generating class will be for the best, we don't have the appropriate time to implement such a feature

Reply:
1 - I'll consider this in the future.